src/include/duckdb/common/serializer/enum_serializer.hpp

```diff
@@ -0,0 +1,1183 @@
+#pragma once
+#include <stdint.h>
+
+#include "duckdb/common/types.hpp"
+#include "duckdb/common/enums/order_type.hpp"
+#include "duckdb/common/enums/tableref_type.hpp"
+#include "duckdb/common/enums/join_type.hpp"
+#include "duckdb/common/enums/joinref_type.hpp"
+#include "duckdb/common/enums/aggregate_handling.hpp"
+#include "duckdb/common/enums/expression_type.hpp"
+#include "duckdb/common/enums/subquery_type.hpp"
+#include "duckdb/common/enums/set_operation_type.hpp"
+
+#include "duckdb/parser/result_modifier.hpp"
+#include "duckdb/parser/query_node.hpp"
+#include "duckdb/parser/expression/window_expression.hpp"
+#include "duckdb/parser/parsed_data/sample_options.hpp"
+
+namespace duckdb {
+
+namespace EnumSerializer {
+// String -> Enum
+template <class T>
+inline T StringToEnum(const char *value) = delete;
+
+// Enum -> String
+template <class T>
+inline const char *EnumToString(T value) = delete;
+
+//-----------------------------------------------
+// OrderType
+//-----------------------------------------------
+template <>
+inline OrderType StringToEnum(const char *value) {
+	if (strcmp(value, "INVALID") == 0) {
```

Comment:
`strcmp` is pretty error prone because it returns `0` on equality, maybe we can add a `StringUtil::Equals` and use that instead?